tests/phpunit/tests/cron.php

```diff
@@ -1112,4 +1112,27 @@ function() {
 		$this->assertSame( 'could_not_set', $unscheduled->get_error_code() );
 	}
 
+	/**
+	 * Tests that a call to wp_schedule_event() on a site without any scheduled events,
+	 * does not result in a PHP deprecation warning on PHP 8.1 or higher.
+	 *
+	 * Warning which we should not see:
+	 * `Deprecated: Automatic conversion of false to array is deprecated`
+	 *
+	 * @ticket 53635
+	 *
+	 * @covers ::wp_schedule_event
+	 */
+	function test_wp_schedule_event_without_cron_option_does_not_throw_warning() {
+		delete_option( 'cron' );
+
+		// Verify that the cause of the error is in place.
+		$this->assertFalse( _get_cron_array(), '_get_cron_array() does not return false' );
+
+		$hook = __FUNCTION__;
+		$ts1  = strtotime( '+10 minutes' );
+
+		// Add an event.
+		$this->assertTrue( wp_schedule_event( $ts1, 'daily', $hook ) );
+	}
```

Comment:
Are you able to test `wp_schedule_single_event()` and `wp_unschedule_event()` they two make assumptions about an array. https://3v4l.org/YoClp/rfc

Doing a quick review of the file, the remaining functions check for either empty or false but I will double check that and confirm. 
 
`wp_schedule_single_event()` currently has an unrelated bug (introduced my me, oops) that casts `_get_cron_array()` without checking it's `false` so it would be good to make sure the deprecation notice doesn't get thrown when this bug is fixed.
https://github.com/WordPress/wordpress-develop/blob/1c5d6899a0c76a4a61833ff796dab60d12ede014/src/wp-includes/cron.php#L121

`wp_unschedule_event()` can throw a notice on 8.1 if the cron array isn't set.

Reply:
@peterwilsoncc Happy to look into this at some point in the future, but both points you make are outside the scope of this particular PR.

This PR addresses one specific PHP 8.1 error and adds test for that particular error. Errors in other functions should be addressed separately. (also see my other PRs)

Reply:
Cool, didn't realise you were doing it function-by-function.

I'll add the `wp_unschedule_event()` notes to the PHP 8.1 trac ticket.

Reply:
@peterwilsoncc Actually going based on errors seen in the PHPUnit logs at this moment and addressing one specific error at a time. This is/was one showing at the top of each PHPUnit run before the tests would even start.

Made a note to look at the other functions too though.

Reply:
Here is how it shows in the last run against `trunk`: https://github.com/WordPress/wordpress-develop/runs/3333385291?check_suite_focus=true#step:20:25

Reply:
@peterwilsoncc I've done a quick search of the `src` to see where `_get_cron_array()` gets called - result:
```
wp-admin/includes/class-wp-media-list-table.php [1]
wp-admin/includes/class-wp-site-health.php [1]
wp-admin/includes/upgrade.php [1]
wp-includes/cron.php [12]
```

I'll verify each of these.

Reply:
Oops, forgot to refresh the tab before committing in https://core.trac.wordpress.org/changeset/51619, and didn't see these latest comments. But it looks like these instances can be addressed separately.

Reply:
@SergeyBiryukov They will be and should (addressed separately), so all is good. 👍🏻

Reply:
> `wp_schedule_single_event()` currently has an unrelated bug (introduced my me, oops) that casts `_get_cron_array()` without checking it's `false` so it would be good to make sure the deprecation notice doesn't get thrown when this bug is fixed.
> 
> https://github.com/WordPress/wordpress-develop/blob/1c5d6899a0c76a4a61833ff796dab60d12ede014/src/wp-includes/cron.php#L121

@peterwilsoncc Is there a Trac ticket open for this bug already ? I've created a test reproducing the bug + a fix which will be PHP cross-version compatible, but as you noted already, this bug is not PHP 8.1 related, so should get its own ticket.

Reply:
Thanks friend, no ticket as yet. I'll try to create it during the next few hours and post a link here.

Reply:
@peterwilsoncc When you do, the PR with the patch is here: https://github.com/WordPress/wordpress-develop/pull/1598

Reply:
> Are you able to test `wp_schedule_single_event()` and `wp_unschedule_event()` they two make assumptions about an array. https://3v4l.org/YoClp/rfc
> 
> `wp_unschedule_event()` can throw a notice on 8.1 if the cron array isn't set.

Actually, the code in `wp_unschedule_event()` is fine in principle, but `_set_cron_array()` is not as it has no input validation, so when `wp_unschedule_event()` passes off to `_set_cron_array()`, an error can occur.

Reply:
@peterwilsoncc I've also now opened a PR to address `_set_cron_array()`: https://github.com/WordPress/wordpress-develop/pull/1599

Reply:
I've now verified all uses of `_get_cron_array()` and created fixes where needed. Aside from the above mentioned PR, a ticket needs to be opened still for `WP_Media_List_Table::prepare_items()` as that, similar to `wp_schedule_single_event()` contains a PHP cross-version (not PHP 8.1 specific) bug.

Reply:
WP_Media_List_Table bug: https://core.trac.wordpress.org/ticket/53949 and PR: https://github.com/WordPress/wordpress-develop/pull/1600